src/app/screens/Transactions/index.tsx

```diff
@@ -0,0 +1,70 @@
+import Container from "@components/Container";
+import TransactionsTable from "@components/TransactionsTable";
+import dayjs from "dayjs";
+import { useCallback, useEffect, useState } from "react";
+import { useTranslation } from "react-i18next";
+import { toast } from "react-toastify";
+import { useSettings } from "~/app/context/SettingsContext";
+import msg from "~/common/lib/msg";
+import { Payment, Transaction } from "~/types";
+
+function Transactions() {
+  const { t } = useTranslation("translation", {
+    keyPrefix: "transactions",
+  });
+
+  const { settings, getFormattedFiat } = useSettings();
+  const [payments, setPayments] = useState<Transaction[]>([]);
+
+  const fetchData = useCallback(async () => {
+    try {
+      const paymentsResponse = await msg.request<{
+        payments: Payment[];
+      }>("getPayments");
+
+      const transactions: Transaction[] = paymentsResponse.payments.map(
```

Comment:
Wondering if it makes sense to abstract this part. We use the same/similar code in different places now.

Reply:
Yes, also the mapping afterwards. I'll see

Reply:
Done